### ISSUE TITLE
Can't read unix-style timestamps

### ISSUE THREAD

**Issue by AndrewM-** [geopaparazzi/geopaparazzi#534]
Each row in each table has a ts field but this is not in a human readable timestamp format.  If the gpap file is opened in QGIS or in Manifold GIS or in some db management utility, I can't read the timestamp.  I know that I can write a function that converts the unix-style time stamp into a text timestamp but I can't implement that easily with my tool chain.  Additionally, I can't import gpap files into ms-access using the sqlite driver by ch-werner.de (only free one I can find), without losing the ts field.  The sqlite driver can't process date data.  

If I could get the timestamp into ms-access, I could automate the merging of gpap data into a single master database using ms-access.  Having an enterprise database with all geopaparazzi field data would then be easy to create and manage.  

Can we have timestamps as text in the following format: "yyyy/mm/dd HH/MM/SS", in addition to the ts field.  Text time stamps are also useful because the do not get ruined by the regional settings that windows uses.

**Comment by brentfraser** [comment]
As a work-around you could create views in the Project SQLite databases for each of the tables, like

    CREATE view v_images AS
    SELECT datetime(ts / 1000, 'unixepoch') ts_string, * FROM images; 

Not great, but is it scriptable, etc.

**Comment by mj10777** [comment]
The use of '/' as a separator for dates should be avoided at all costs.

The '/' as a date separator is mainly used in the Anglo-Saxon areas
* British: 'DD/MM/YYYY'
* others: 'MM/TT/YYYY'

and was the cause of much confusion when transferring data in a CSV form in the 1990's.
The cause of this confusion should be clear and yet 

To avoid this, an ISO Norm (8601) was created and is used in sqlite3 as the default Date-Format:
* 'YYYY-MM-DD'

Spatialite uses '2018-06-26T14:09:49.988Z' as a timestamp in **spatialite_history**

https://en.wikipedia.org/wiki/ISO_8601

---

The use of integer values for timestamps
* unix: based on '1960-01-01'
* dos: based on '1970-01-01'

saves space and is swifter which is the main reason for its use.

A software that does not support this, should not be used.

**Comment by brentfraser** [comment]
The SQLite  datetime function I used above  gives something close to ISO 8601:

    2017-08-05 16:11:30

so it might produce a data string acceptable to QGIS etc.

**Comment by mj10777** [comment]
SELECT strftime( '%Y-%m-%dT%H:%M:%SZ', datetime( 'now' ))

Will return the UTM time (a 'Z' should be added).

SELECT strftime( '%Y-%m-%dT%H:%M:%S', datetime( 'now','localtime' ))

Will return the local time.

SELECT strftime( '%Y-%m-%dT%H:%M:%SZ',datetime(1092941466, 'unixepoch'));

Will return the unix-timestamp in a form that any software should be able to read.

**Comment by AndrewM-** [comment]
Thanks guys.  I am using dbeaver to make the changes.  I also tried adding data and time fields to the tags.json file and having a no edit section on the form as a means of recording date and time, but the user still has to open the control to save the date or time.

**Comment by brentfraser** [comment]
As @mj10777 pointed out you can create an ISO 8601 date-time string in a query.  Just remember to divide the "ts" value stored in the Geopaparazzi tables by 1000 as it stores milliseconds, not seconds.

**Comment by AndrewM-** [comment]
It is great that the ts value is in milliseconds.  It is still too hard for casual users to create human readable time stamps and I still recommend adding a new column to the notes, photos and bookmarks tables with iso timestamps.  Yesterday, I spent hours trying to find a entry level GIS user way of creating human readable timestamps in QGIS.  The best that I have been able to come up with is to write a small application that files the gpap file on the server and at the same time shells to the sqlite command line to run sql commands to add and populate time stamp fields (in development).  There are no native QGIS functions for processing unixepoch timestamps in the normal field calculator although it may be possible to script the python advanced field calculator to do this process (got halfway there).

**Comment by AndrewM-** [comment]
I found a library that allows me to access gpap files from within ms-access.  I had to manually edit the library to solve the timestamp issue.  I had to remap the sqlite integer type to ms-access doubles as the time stamps from Geopaparazz are too long for ms-access's long integer type.  Converting from a double to a time stamp is easy and people can look it up.  

https://github.com/govert/SQLiteForExcel - a link to the ms-access version is on the same page.
